src/coprocessor/readpool_impl.rs

```diff
@@ -83,15 +83,38 @@ pub fn build_read_pool_for_test<E: Engine>(
     configs
         .into_iter()
         .map(|config| {
-            let engine = Arc::new(Mutex::new(engine.clone()));
+            let eng = Arc::new(Mutex::new(engine.clone()));
             Builder::from_config(config)
-                .after_start(move || set_tls_engine(engine.lock().unwrap().clone()))
+                .after_start(move || after_start(&eng))
                 .before_stop(|| destroy_tls_engine::<E>())
                 .build()
         })
         .collect()
 }
 
+#[cfg(target_os = "linux")]
+fn after_start<E: Engine>(engine: &Arc<Mutex<E>>) {
+    set_tls_engine(engine.lock().unwrap().clone());
+    unsafe {
+        // Set to lowest prio.
+        *libc::__errno_location() = 0i32;
+        if libc::nice(100) == -1 && *libc::__errno_location() != 0 {
```

Comment:
I used to think that `nice` was only be allowed to set `priority` of `process`, instead of `thread`.  I'm not sure that what you do is right. Why not just use `pthread_setschedparam`?